examples/step-60/step-60.cc

```diff
@@ -0,0 +1,1077 @@
+  {
+    TimerOutput::Scope timer_section(monitor, "Output results");
+
+    DataOut<spacedim> embedding_out;
+
+    std::ofstream embedding_out_file("embedding.vtu");
+
+    embedding_out.attach_dof_handler(*space_dh);
+    embedding_out.add_data_vector(solution, "solution");
+    embedding_out.build_patches(parameters.embedding_space_finite_element_degree);
+    embedding_out.write_vtu(embedding_out_file);
+
+    // The only difference between the two output routines, is that in the
+    // second case, we want to output the data on the current configuration, and
+    // not on the reference one. This is possible by passing the actual
+    // embedded_mapping to the DataOut::build_patches function. The mapping will
+    // take care of outputting the result on the actual deformed configuration.
+
+    DataOut<dim, DoFHandler<dim,spacedim> > embedded_out;
+
+    std::ofstream embedded_out_file("embedded.vtu");
+
+    embedded_out.attach_dof_handler(*embedded_dh);
+    embedded_out.add_data_vector(lambda, "lambda");
+    embedded_out.add_data_vector(embedded_value, "g");
+    embedded_out.build_patches(*embedded_mapping,
+                               parameters.embedded_space_finite_element_degree);
+    embedded_out.write_vtu(embedded_out_file);
+  }
+
+  // Nothing special here, except that we check if parsing was done before
+  // we actually attempt to run our program.
+
+  template<int dim, int spacedim>
+  void DistributedLagrangeProblem<dim, spacedim>::run()
+  {
+    AssertThrow(parameters.initialized,
+                ExcNotInitialized());
+    deallog.depth_console(parameters.verbosity_level);
+
+    setup_grids_and_dofs();
+    setup_coupling();
+    assemble_system();
+    solve();
+    output_results();
+  }
+}
+
+
+
+int main(int argc, char **argv)
+{
+  try
+    {
+      using namespace dealii;
+      using namespace Step60;
+
+      const unsigned int dim=1, spacedim=2;
+
+      // Differently to what happens in other tutorial programs, here we use
+      // ParameterAcceptor style of initialization, i.e., all objects are first
+      // constructed, and then a single call to the static method
+      // ParameterAcceptor::initialize is issued to fill all parameters of the
+      // classes that are derived from ParameterAcceptor.
+      //
+      // We check if the user has specified a parameter file name to use when
+      // the program was launched. If so, try to read that parameter file,
+      // otherwise, try to read the file "parameters.prm".
+      //
+      // If the parameter file that was specified (implicitly or explicitly)
+      // does not exist, ParameterAcceptor::initialize will create one for you,
+      // and exit the program.
+
+      DistributedLagrangeProblem<dim, spacedim>::DistributedLagrangeProblemParameters parameters;
+      DistributedLagrangeProblem<dim, spacedim> problem(parameters);
+      std::string parameter_file = "parameters.prm";
+      if (argc > 1)
+        parameter_file = argv[1];
```

Comment:
I think this would be easier to read if you did
```
  std::string parameter_file;
  if (argc > 1)
    parameter_file = argv[1];
  else
     parameter_file = "parameters.prm";
```
(Single assignment style.)